query_optimizer/PhysicalGenerator.cpp

```diff
@@ -109,6 +114,13 @@ P::PhysicalPtr PhysicalGenerator::optimizePlan() {
   } else {
     rules.emplace_back(new SwapProbeBuild());
   }
+  if (FLAGS_reorder_columns) {
+    // NOTE(jianqiao): This optimization relies on the fact that the intermediate
+    // relations all have SPLIT_ROW_STORE layouts. If this fact gets changed, the
+    // optimization algorithm may need to be updated and the performance impact
+    // should be re-evaluated.
```

Comment:
Currently this cannot be checked because the temporary relations' layouts are generated in `ExecutionGenerator` when they get created.

In the future we will reason about the layouts during query optimization and add corresponding data structures.